src/Form/Traits/AllowedLinkClassesTrait.php

```
@@ -0,0 +1,124 @@
+<?php
+
+namespace SilverStripe\LinkField\Form\Traits;
+
+use InvalidArgumentException;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\LinkField\Services\LinkTypeService;
+
+/**
+ * Trait to manage which Link type can be added to LinkField form field.
+ * This trait is used in LinkField and MultiLinkField classes.
+ */
+trait AllowedLinkClassesTrait
+{
+    private array $allowed_types = [];
+
+    /**
+     * Set allowed types for LinkField
+     * @param string[] $types
+     */
+    public function setAllowedTypes(array $types): static
```

What's our thinking when an empty array is passed to this method?
- Do we want to throw an exception?
- Do we want to unset any restriction and go back to allowing all Link types?

---

I put my comment here https://github.com/silverstripe/silverstripe-linkfield/pull/137#issuecomment-1859327356.
I think empty array is invalid argument.